src/tutorial/testing/src/main.rs

```diff
@@ -22,7 +22,7 @@ fn main() -> Result<(), ExitFailure> {
     Ok(())
 }
 
-fn find_matches<W: std::io::Write>(content: &str, pattern: &str, writer: &mut W) {
+fn find_matches(content: &str, pattern: &str, mut writer: impl std::io::Write) {
```

Comment:
do you think it is a good idea to introduce both traits & impl trait at the same time to a beginner? I think we can avoid impl Trait here to keep the barrier low or mention an aside about the feature

Reply:
I actually thought the same thing, and that's why I wrote the named type parameter in the first draft. But this change was made specifically because a beginner was confused by the `<W: std::io::Write>` syntax, as they had no way of telling what it does. `impl Write` is more obvious in that it refers to something you know from `impl` blocks. Maybe we are just too used to seeing the other syntax?

Note: This is not `impl Trait` as you know it from return types. It's just an alternative to the other syntax that was used before.